src/game/engine/vessel.cpp

```diff
@@ -34,20 +34,16 @@ VesselClass::VesselClass(VesselType type, HousesType house) :
     m_Class(g_VesselTypes.Ptr(type)),
     m_ToSelfRepair(false),
     m_Repairing(false),
-    m_TransportDoorTimer(),
-    m_SubmergeTimer(),
+    m_TransportDoorTimer(0),
+    m_SubmergeTimer(0),
     m_SecondaryTurretFacing()
 {
-    if (m_OwnerHouse != nullptr) {
-        m_OwnerHouse->Tracking_Add(this);
-    }
+    m_OwnerHouse->Tracking_Add(this);
 
-    if (What_Type() != VESSEL_NONE) {
-        m_Bit2_16 = !(Class_Of().Is_Two_Shooter() == true);
-        m_Cloakable = Class_Of().Is_Cloakable();
-        m_Ammo = Class_Of().Get_Ammo();
-        m_Health = Class_Of().Get_Strength();
-    }
+    m_Ammo = Class_Of().Get_Ammo();
+    m_Bit2_16 = !(Class_Of().Is_Two_Shooter() == true);
+    m_Health = Class_Of().Get_Strength();
+    m_Cloakable = Class_Of().Is_Cloakable();
 }
 
 VesselClass::VesselClass(const VesselClass &that) :
```

Comment:
This function needs newlines. Also, you don't need to put newlines between every statement. There is no need for them between the Tracking_Remove, Destroy_Cargo and Limbo calls for example. Looks like Destroy_Cargo has a typo that needs fixing as well.